clang/lib/CIR/Dialect/IR/CIRTypes.cpp

```diff
@@ -309,6 +309,37 @@ RecordType::computeStructAlignment(const mlir::DataLayout &dataLayout) const {
   return recordAlignment;
 }
 
+uint64_t RecordType::getElementOffset(const ::mlir::DataLayout &dataLayout,
+                                      unsigned idx) const {
+  assert(idx < getMembers().size() && "access not valid");
+
+  // All union elements are at offset zero.
+  if (isUnion() || idx == 0)
+    return 0;
+
+  assert(isComplete() && "Cannot get layout of incomplete records");
+  assert(idx < getNumElements());
+  llvm::ArrayRef<mlir::Type> members = getMembers();
+
+  unsigned offset = 0;
+
+  for (unsigned i = 0, e = idx; i != e; ++i) {
```

Comment:
Is there good reason to not use range-for here? Something like:


`for (mlir::Type ty : std::make_range(members.begin(), std::next(members.begin(), idx)))` (though there is perhaps some ADT stuff to do a better job of this)

Reply:
`auto i : llvm::seq<unsigned>(0, idx))` could be a good one

Reply:
> Is there good reason to not use range-for here? Something like:
> 
> `for (mlir::Type ty : std::make_range(members.begin(), std::next(members.begin(), idx)))` (though there is perhaps some ADT stuff to do a better job of this)

Given that it wasn't ending at the end, it seemed just as well to iterate like this. Is there an advantage to the range-for in this case, given the extra calls to set it up?

Reply:
Typically clang prefers range-for for readability and avoiding the 'error-prone-ness' of an index.  Readability is because it assures that we are only visiting anything 'once', and the index is unchanged between iterations.  Typically when I see one of these types of loops, it means "funny-business is happening in the body, so look closely".